lib/vsc/filesystem/posix.py

```diff
@@ -453,6 +441,23 @@ def populate_home_dir(self, user_id, group_id, home_dir, ssh_public_keys):
                 self.log.raiseException("Cannot change ownership of file %s to %s:%s" %
                                         (f, user_id, group_id), PosixOperationError)
 
+    def _deploy_dot_file(self, path, filename, user_id, contents):
+        """
+        Deploy a .dot file
+
+        Checks for symlinks, only overwrites these if the target is missing.
+        """
+
+        if os.path.exists(path):
+            self.log.info("%s already exists for user %s. Not overwriting.", filename, user_id)
+        else:
+            self.log.info("%s not found for user %s. Writing default.", filename, user_id)
+            if os.path.islink(path):
+                self.log.info("%s is symlinked to non-existing target %s ", filename, os.path.realpath(path))
```

Comment:
where do you check that the target doesn't exist?

Reply:
Before this gets called.

Reply:
good point though.

Reply:
Right, os.path.exisists ` Returns False for broken symbolic links.` that's makes more sense.